crates/staged-sync/tests/sync.rs

```diff
@@ -0,0 +1,310 @@
+
+        tracing::info!(genesis_hash = ?chainspec.genesis_hash, "genesis hash");
+        tracing::info!(tip_hash = ?tip_hash, "tip hash");
+        tracing::info!(tip_number = ?tip.number, "tip number");
+
+        // === initialize reth networking stack ===
+
+        let secret_key = SecretKey::new(&mut rand::thread_rng());
+        let (reth_p2p, reth_disc) = unused_tcp_udp();
+        tracing::info!(
+            %reth_p2p,
+            %reth_disc,
+            "setting up reth networking stack in sync test"
+        );
+
+        let config = NetworkConfig::<Arc<NoopProvider>>::builder(secret_key)
+            .listener_addr(reth_p2p)
+            .discovery_addr(reth_disc)
+            .chain_spec(chainspec.clone())
+            .build(Arc::new(NoopProvider::default()));
+
+        let network = NetworkManager::new(config).await.unwrap();
+        let handle = network.handle().clone();
+
+        // initialize db
+        let reth_temp_dir = tempfile::tempdir().expect("should be able to create reth data dir");
+        let db = Arc::new(init_db(reth_temp_dir.path()).unwrap());
+
+        // initialize consensus
+        let consensus = Arc::new(TestConsensus::default());
+
+        // set up downloaders
+        let fetch_client = Arc::new(network.fetch_client());
+
+        let bodies_downloader = ConcurrentDownloaderBuilder::default()
+            .build(fetch_client.clone(), consensus.clone(), db.clone());
+        let headers_downloader = LinearDownloadBuilder::default()
+            .build(consensus.clone(), fetch_client);
+
+        // initialize pipeline
+        let mut reth: Pipeline<Env<WriteMap>, NoopSyncStateUpdate> = Pipeline::builder()
+            .add_stages(DefaultStages::new(consensus, headers_downloader, bodies_downloader))
+            .build();
+
+        // start reth then manually connect geth
+        let pipeline_handle = tokio::task::spawn(async move {
+            reth.run(db).await.unwrap();
+        });
+
+        tokio::task::spawn(network);
+
+        // create networkeventstream to get the next session established event easily
+        let mut events = NetworkEventStream::new(handle.event_listener());
+        let geth_socket = SocketAddr::new([127, 0, 0, 1].into(), geth_p2p_port);
+
+        // === ensure p2p is active ===
+
+        // get the peer id we should be expecting
+        let geth_peer_id: PeerId = provider.peer_id().await.unwrap();
+
+        // add geth as a peer then wait for `PeerAdded` and `SessionEstablished` events.
+        handle.add_peer(geth_peer_id, geth_socket);
+
+        // wait for the session to be established
+        let _peer_id = events.peer_added_and_established().await.unwrap();
+
+        tracing::info!("waiting for pipeline to finish");
+        pipeline_handle.await.unwrap();
+
+        drop(clique);
+
+        // cleanup (delete the data_dir at dir_path)
+        fs::remove_dir_all(dir_path).await.unwrap();
+    })
+    .await
+    .unwrap();
```

Comment:
I believe this should be unnecessary because tempdir already does this on drop

Reply:
ah I had to do this because I was using `into_path`, which requires manual deletion, but switching to `path` works and does not require doing this